draft-ietf-core-sid.md

```diff
@@ -447,11 +447,18 @@ assignments can then be stored in a ".sid" file. For
 example on how this could be achieved, please refer to {{sid-lifecycle-ex}}.
 
 Items introduced by a new revision of a YANG module are added to the list of SIDs already assigned.
-When this is done development of a new protocol document it may be necessary to make provisional assignments.
+When this is done during development of a new protocol document, it may be necessary to make provisional assignments.
 They may get changed, revised or withdrawn during the development of a new standard.
-These provisional assignments are marked with a status of "unstable".
+These provisional assignments are marked with a status of "unstable",
+so that they can be removed and the SID number possibly be reassigned
+for a different YANG schema name/path later during development.
 When the specification is advanced to a final document, then
 the assignment is marked with a status of "stable".
+During the period of development, two variants of the SID file should
+be made available by the tooling involved in that development: (1) a
+"published" SID file with the existing stable SID assignments only, as
+well as (2) an "unpublished" SID file that also contains the unstable
+SID assignments.
 
```

Comment:
> This new paragraph raises concerns:
> 
> * should instead of SHOULD

I used should because this is not an interoperability requirement.

> * there may not be a published version yet

I fixed the language to "During a period of development starting from a published
specification"

> * the organization should decide how unpublished modules and SIDs are made available

Obviously -- hence "should"

> * It does not make sense to create 2 versions of the SID file for the unpublished YANG module

See fix above.

> * The user should select the SID file that goes with the published YANG module

A user participating in the development will use the development SID files.
A user non participating in the development should use the previous generation of stable files (module and SID).

> * For IETF YANG there are ietf/RFC and ietf/DRAFT folders and the YANG versions in each do not overlap
> * The revision date will be older and different for the ietf/RFC version

Indeed.  Does the text need to discuss this?

I pushed a change, please have a look.

Reply:
I do not agree that it is a good idea to create or use a SID file that has the unstable SIDs removed.
The YANG file may not function properly with this subset of objects.
There is also no need at all for this SID file since the last published version is
supposed to match the contents of the special pruned SID file.
Two SID files per YANG file is not useful but it will confuse people.

Reply:
If the developer can find the YANG work-in-progress then they should find the full SID file
for that YANG file as well.

Example: github IETF

     ietf/RFC/foo@2020-01-01.yang    (somewhere is foo@2020-01-01.sid)

     ietf/DRAFT/foo@2023-01-01.yang  (somewhere is foo@2023-01-01.sid)

Why would I want a SID file for YANG(2023-01-01) if I am using YANG(2020-01-01)?